Neos.Flow/Classes/ObjectManagement/Configuration/ConfigurationBuilder.php

```diff
@@ -118,14 +119,32 @@ public function buildObjectConfigurations(array $availableClassAndInterfaceNames
                 if (isset($rawObjectConfiguration['className'])) {
                     $rawObjectConfiguration = $this->enhanceRawConfigurationWithAnnotationOptions($rawObjectConfiguration['className'], $rawObjectConfiguration);
                 }
-                $newObjectConfiguration = $this->parseConfigurationArray($objectName, $rawObjectConfiguration, 'configuration of package ' . $packageKey . ', definition for object "' . $objectName . '"', $existingObjectConfiguration);
+                if (strpos($objectName, ':') !== false) {
+                    if (!isset($rawObjectConfiguration['factoryObjectName'])) {
```

Comment:
Clever approach. I do like it even though it's hard to understand what is actually done here. Maybe a few doc comments can help that.

Reply:
Yes, I'll extract that to a method and add tests for it